server/src/main/scala/org/http4s/server/middleware/Caching.scala

```diff
@@ -0,0 +1,111 @@
+package org.http4s.server.middleware
+
+import cats._
+import cats.implicits._
+import cats.effect._
+import cats.data._
+import org.http4s._
+import org.http4s.util.CaseInsensitiveString
+import org.http4s.headers.{Date => HDate, _}
+import scala.concurrent.duration._
+
+object Caching {
+  def `no-store`[G[_]: Monad: Clock, F[_], A](
+      http: Kleisli[G, A, Response[F]]): Kleisli[G, A, Response[F]] =
+    Kleisli { a: A =>
+      for {
+        resp <- http(a)
+        now <- HttpDate.current[G]
+      } yield {
+        val headers = List(
+          `Cache-Control`(
+            NonEmptyList.of[CacheDirective](
+              CacheDirective.`no-store`,
+              CacheDirective.`private`(),
+              CacheDirective.`no-cache`(),
+              CacheDirective.`max-age`(0.seconds)
+            )),
+          Header("Pragma", "no-cache"),
+          HDate(now),
+          Expires(HttpDate.Epoch) // Expire at the epoch for no time confusion
+        )
+        resp.putHeaders(headers: _*)
+      }
+    }
+
+  object Helpers {
```

Comment:
Does this need to be public?  If it does, I might consider moving the two helpers herein to the top-level.  Things nested in `Helpers` or `Utils` or `bits` tend to not be discoverable.

Reply:
The thing I was hoping to achieve was to allow those who used `caching` to have a way to get the tools we use rather than recreate them, but only leave the middlewares exposed at the top level.

Reply:
Okay, fine with me as is.